ouroboros-network/src/Ouroboros/Network/Testing/ConcreteBlock.hs

```diff
@@ -272,6 +276,38 @@ fixupBlockHeader pb b = b'
       headerBodyHash = headerBodyHash b
     }
 
+-- |
+-- Like @'fixupBlock'@ but it will preserve the first block instead of using
+-- @GenesisHash@ and @BlockNo 1@.
+--
+fixupBlockHeader' :: (HasHeader block, HeaderHash block ~ HeaderHash Block)
+                 => Maybe block -> BlockHeader -> BlockHeader
+fixupBlockHeader' pb b = b'
```

Comment:
Would this not also work for chains that start from genesis? Perhaps we don't need two variants and this variant covers both?